### ISSUE TITLE
[QUESTIONS] change attribute value that is generated through trait

### ISSUE THREAD

**Issue by HarshOB** [artf/grapesjs#996]
Now, let's consider that I wrote **new** into the trait value but in attribute **:items**, I want to append prefix like **item.text.** to that value So, my final attribute **:items** become **item.text.new**. Is that possible? 

![screenshot from 2018-04-03 12-52-00](https://user-images.githubusercontent.com/28950214/38235247-0c5d655e-373e-11e8-81cd-0cdad9eef9c3.png)

**Comment by artf** [comment]
You should be able to achieve it via a [custom trait](https://github.com/artf/grapesjs/wiki/Traits#define-new-trait-type), eg. by overriding [`getValueForTarget` method](https://github.com/artf/grapesjs/blob/dev/src/trait_manager/view/TraitView.js#L46)

**Comment by HarshOB** [comment]
I tried creating custom trait and override a method but didn't see any attribute generated. can you please look at this code and correct me if there's any error. When I write data in traits it should generate attribute as  ':text=item.text.data' at datatextfield.

```
 getInputEl: function() {
    if (!this.inputEl) {
      var input = document.createElement('textarea');
      input.value = this.target.get('content');
      this.inputEl = input;
    }
    return this.inputEl;
  },

  /**
   * Triggered when the value of the model is changed
   */
  onValueChange: function () {
    this.target.set('content', this.model.get('value'));
  },
  
  getValueForTarget: function() {
      console.log("inside getValueForTargetss")
    return this.model.get( 'item.text' + 'value');
  }
});
```


![screenshot from 2018-04-12 10-21-21](https://user-images.githubusercontent.com/28950214/38657005-a504f80e-3e3b-11e8-9346-a6185ffef5a5.png)

**Comment by artf** [comment]
I'd say it's enough just this
```js
getValueForTarget: function() {
    return 'item.text' + this.model.get('value');
  }
```

**Comment by HarshOB** [comment]
Thanks, @artf for this help. I got another problem (I have multiple pages) when I set a value of trait using method overloading then it's value is not persistent in trait textarea. like if I typed "id" in textarea then click on other page and come back to same page and then if I open settings and saw then it's value is automatically cleared from the textarea but it is still in attribute. How can I set value in traits that is present in attribute?
